calico/src/main/scala/calico/dsl.scala

```diff
@@ -329,15 +329,28 @@ object Children:
 
   given [F[_], E <: dom.Element](using F: Async[F]): Modifier[F, E, Modified[F]] with
     def modify(children: Modified[F], e: E) =
-      Stream
-        .resource(Hotswap.create[F, Unit])
-        .flatMap { hs =>
-          children.cs.evalMap(c => hs.swap(c.evalMap(c => F.delay(e.replaceChildren(c*)))))
-        }
+      for
+        hs <- Hotswap.create[F, List[dom.Node]]
+        placeholder <- Resource.make(
+          F.delay(e.appendChild(dom.document.createComment("")))
+        )(
+          placeholder => F.delay(e.removeChild(placeholder))
+        )
+        _ <- children.cs.evalScan(List.empty[dom.Node])((prevChildren, c) => {
+          val sibling = prevChildren.headOption.orNull
+          hs.swap(c.evalMap { c =>
+            F.delay {
+              prevChildren.foreach(e.removeChild)
+              c.foreach(e.insertBefore(_, placeholder))
+              c
+            }
```

Comment:
This seems correct to me. I wonder what the performance is like vs `replaceChildren`. But I'm also not sure if it matters in practice.

Reply:
Maybe this will be more efficient?
```suggestion
            F.delay {
              prevChildren.foreach(e.removeChild)
              placeholder.after(c: _*)
              c
            }
```

Reply:
comment nodes don't extend `Element` so it doesn't have an `after` method. Is there a different element that might work better?

Reply:
Oh darn, nevermind then :) I think comment is the right choice.